grype/db/v6/store.go

```diff
@@ -55,18 +56,38 @@ func newStore(cfg Config, empty, writable bool) (*store, error) {
 		return nil, fmt.Errorf("failed to open db: %w", err)
 	}
 
+	metadataStore := newDBMetadataStore(db)
```

Comment:
I don't understand this change, probably because I don't have a good enough mental model of the grype db schema v6 database lifecycle. Can you help me understand?

Reply:
the goal of this change is to get the DB metadata from the DB since it is used downstream in the vulnerability decorator store (new) to determine client-vs-db capabilities are supported.

Lifecycle wise, there are two times this store is used: when creating a new DB from scratch (in grype-db) or when we are matching packages with vulnerabilities (read only in grype). The first case we start with an empty DB and start adding entries, in which case we need to first write the DB metadata in order to read it. That's why there is a "check if empty is set (which means truncate the DB in preparation for writes) and then set the metadata before getting it"... the read only case will not hit this condition and will simply read the metadata from the DB.

Reply:
That makes sense. The indentation change in the return statement was hiding `vulnerabilityDecoratorStore: newVulnerabilityDecoratorStore(db, bs, dbVersion),` from me, so I didn't understand that previously, we didn't need to get the metadata during this function, and now we do.